src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Tests.cs

```diff
@@ -125,28 +127,241 @@ public class TarReader_File_Tests : TarReader_File_Tests_Base
+        public void ReadCopiedDataStreamOfExternalAssetsGoLang(string testCaseName) =>
+            VerifyDataStreamOfTarUncompressedInternal("golang_tar", testCaseName, copyData: true);
+
+        [Theory]
+        [MemberData(nameof(GetNodeTarTestCaseNames))]
+        public void ReadCopiedDataStreamOfExternalAssetsNode(string testCaseName) =>
+            VerifyDataStreamOfTarUncompressedInternal("node-tar", testCaseName, copyData: true);
+
+        [Theory]
+        [MemberData(nameof(GetRsTarTestCaseNames))]
+        public void ReadCopiedDataStreamOfExternalAssetsRs(string testCaseName) =>
+            VerifyDataStreamOfTarUncompressedInternal("tar-rs", testCaseName, copyData: true);
+
+        [Fact]
+        public void Throw_FifoContainsNonZeroDataSection()
+        {
+            using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, "golang_tar", "hdr-only");
+            using TarReader reader = new TarReader(archiveStream);
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.NotNull(reader.GetNextEntry());
+            Assert.Throws<FormatException>(() => reader.GetNextEntry());
+        }
+
         [Fact]
-        public void Read_NodeTarArchives_Successfully()
+        public void Throw_SingleExtendedAttributesEntryWithNoActualEntry()
         {
-            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", "node-tar");
-            foreach (string file in Directory.EnumerateFiles(nodeTarPath, "*.tar", SearchOption.AllDirectories))
+            using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, "golang_tar", "pax-path-hdr");
+            using TarReader reader = new TarReader(archiveStream);
+            Assert.Throws<EndOfStreamException>(() => reader.GetNextEntry());
+        }
+
+        [Theory]
+        [InlineData("tar-rs", "spaces")]
+        [InlineData("golang_tar", "v7")]
+        public void AllowSpacesInOctalFields(string folderName, string testCaseName)
+        {
+            using MemoryStream archiveStream = GetTarMemoryStream(CompressionMethod.Uncompressed, folderName, testCaseName);
+            using TarReader reader = new TarReader(archiveStream);
+            TarEntry entry;
+            while ((entry = reader.GetNextEntry()) != null)
             {
-                using FileStream sourceStream = File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read);
-                using var reader = new TarReader(sourceStream);
+                AssertExtensions.GreaterThan(entry.Checksum, 0);
+                AssertExtensions.GreaterThan((int)entry.Mode, 0);
```

Comment:
Is there a test where we expext zero? i.e: when the bytes passed into `ParseOctal` are all leading spaces.